code/game/objects/items/dna_probe.dm

```diff
@@ -27,50 +27,60 @@
 	var/list/stored_dna_plants = list()
 	///List of all Human DNA scanned with this sampler.
 	var/list/stored_dna_human = list()
+	///weak ref to the dna vault
+	var/datum/weakref/dna_vault_ref
```

Comment:
you don't justify why you're changing this behavior in your PR, and also it breaks the space dragon final objective.

Reply:
The dna vault used to loop through the probe's dna list and add it to its database. I wanted the probe to check for duplicate directly so the entire list can be added without for loops. Also guh i need to address dragon stuff